src/SFML/Audio/SoundFileReaderFlac.cpp

```diff
@@ -142,10 +138,12 @@ namespace
                         break;
                 }
 
+                // If there's room in the output buffer, copy the sample there
                 if (data->remaining > 0)
                 {
-                    // There's room in the output buffer, copy the sample there
-                    *data->buffer++ = sample;
+                    // If there's no output buffer, it means that we are seeking, so just skip the output write
+                    if (data->buffer)
+                        *data->buffer++ = sample;
                     data->remaining--;
```

Comment:
Ok, I understand it better now. To me, the cleanest way to handle this is to do nothing but add an assert before dereferencing the pointer. Adding runtime checks for stuff that's not supposed to happen, makes it look like it _may_ happen. Adding an assert, on the other hand, explicitly indicates that there's a programming error if this happens.

And yes, you can add a check for `data->buffer` in the surrounding `if`, and I'd also move the new comment about seeking above it.